src/main/java/com/rewedigital/composer/composing/FetchContext.java

```diff
@@ -0,0 +1,46 @@
+package com.rewedigital.composer.composing;
+
+import static java.util.Objects.requireNonNull;
+
+import java.time.Duration;
+import java.util.Optional;
+
+/**
+ * A simple parameter object for {@link ContentFetcher}s.
+ */
+public class FetchContext {
+
+    private final String path;
+    private final String fallback;
+    private final Optional<Duration> ttl;
+
+    private FetchContext(final String path, final String fallback, final Optional<Duration> ttl) {
+        this.path = path;
+        this.fallback = fallback;
+        this.ttl = requireNonNull(ttl);
+    }
+
+    /**
+     * Builds a simple parameter object for {@link ContentFetcher}s.
+     *
+     * @param path to fetch from.
+     * @param fallback the fallback returned in case of an error.
+     * @param ttl how long the fetch should take.
```

Comment:
I wonder if `ttl` is the right name: shouldn't it be `timeout` or something like that?

Reply:
Let's discuss that @work.